v2/devdoc/log_context_property_type_wchar_t_ptr_requirements.md

```diff
@@ -0,0 +1,95 @@
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy` shall succeed and return 0. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).free
+
+```c
+typedef void (*LOG_CONTEXT_PROPERTY_TYPE_FREE)(void* value);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_011: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).free` shall return. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type
+
+```c
+typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_012: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type` shall return the property type `LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr`. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)
+
+```c
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)` initializes the memory at `dst_value` with the `wprintf` style formatted string given by `format` and the arguments in `...`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_013: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_014: [** If `format` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_015: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)` shall initialize by calling `vswprintf` the memory at `dst_value` with the `wprintf` style formatted wchar_t string given by `format` and the arguments in `...`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_016: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)` shall succeed and return 0. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_017: [** If formatting the string fails, `LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)` shall fail and return a non-zero value. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)
+
+```c
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const char* format, ...);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)` returns the amount of memory needed in number of wide-characters to store the `printf` style formatted string given by `format` and the arguments in `...`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_018: [** If `_vsnwprintf` fails, `LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)` shall return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_019: [** Otherwise, on success, `LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)` shall return the amount of memory needed in number of wide-chracters to store the `wprintf` style formatted `wchar_t` string given by `format` and the arguments in `...`. **]**
```

Comment:
>wprintf [](http://example.com/codeflow?start=222&length=7)

`wprintf` needs `wchar_t` argument. 

Line 88 lists `const char* format`. Which one is correct?

Reply:
wchar_t is correct, modified line 88